mgdapi-1346 - update to C03 test

MStokluska opened the pull request:
# Description
Jira https://issues.redhat.com/browse/MGDAPI-1346

## Type of change

Updated the alert mechanism works test to reduce the time it takes for the test to complete.
Also, updated the tests output (added some successful messages):
![image](https://user-images.githubusercontent.com/43473912/112643566-67778480-8e3c-11eb-85b9-d3369b0ef3c5.png)


## Verification
Verify that the e2e tests on this PR passed, or run locally

MStokluska commented:
/test e2e

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test e2e

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test e2e

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test e2e

Boomatang commented on test/common/alerts_mechanism.go:
```diff
@@ -31,73 +26,79 @@ type alertManagerConfig struct {
-	err = getThreescaleAlertState(ctx)
+	err = getKeycloakAlertState(ctx)
 	if err != nil {
 		t.Fatal("failed to get threescale alert state", err)
```
```suggestion
		t.Fatal("failed to get keycloak alert state", err)
```

Boomatang commented on test/common/alerts_mechanism.go:
```diff
@@ -31,73 +26,79 @@ type alertManagerConfig struct {
 		t.FailNow()
 	}
 
-	// scale down Threescale operator and product pods and verify that threescale alert is firing
```
Update comment
`scale down keycloak operator and product pods and verify that keycloak alert is firing`

Boomatang commented on test/common/alerts_mechanism.go:
```diff
@@ -31,73 +26,79 @@ type alertManagerConfig struct {
-	if threescaleAlertsFiring {
-		t.Log("3Scale alerts firing unexpectedly")
+	if keycloakAlertsFiring {
+		t.Log("Keycloak alerts firing unexpectedly")
```
If I am understanding correctly at this point we don't expect to see alerts firing. Something that might be more clear in the logs is `Keycloak alerts firing unexpectedly before deployments are scaled down`

Boomatang commented on test/common/alerts_mechanism.go:
```diff
@@ -31,73 +26,79 @@ type alertManagerConfig struct {
+	t.Log("Scaling down keycloak operator deployment")
+	scaleDeployment(t, keycloakOperatorDeploymentName, 0, ctx.KubeClient)
+
+	t.Log("Keycloak alerts are not firing - performing tests")
```
To be more accurate in the logs this statement would need to go before the `t.Log("Scaling down keycloak operator deployment")` on line 68

MStokluska replied:
This is addressed by line @68

MStokluska replied:
I moved the scaleDeployment into perform tests instead. Thanks!

Boomatang commented on test/common/alerts_mechanism.go:
```diff
@@ -31,73 +26,79 @@ type alertManagerConfig struct {
 	}
+	t.Log("Keycloak operator deployment scaled back up")
 
 	// verify that threescale alert is not firing
```
update comment

Boomatang commented on test/common/alerts_mechanism.go:
```diff
@@ -31,73 +26,79 @@ type alertManagerConfig struct {
 	}
 
+	t.Log("Scaling up keycloak operator deployment")
+	scaleDeployment(t, keycloakOperatorDeploymentName, originalOperatorReplicas, ctx.KubeClient)
```
This function needs to happen even if the test fails at an earlier stage such as the error check on line 73. Please add this as a deferred function.

Boomatang commented on test/common/tests.go:
```diff
@@ -95,9 +96,7 @@ var (
 
-	FAILURE_TESTS = []TestCase{
-		/*FLAKY*/ {"C03 - Verify that alerting mechanism works", TestIntegreatlyAlertsMechanism},
-	}
+	FAILURE_TESTS = []TestCase{}
```
Why was this moved out of the failure tests. This is not a happy path test as it test functionally of a install that is not in a healthy state.

MStokluska replied:
Reason for this is that I had a different understanding of what "FAILURE_TESTS" are. Thanks for explaining what it is!

MStokluska replied:
I've also updated the prow e2e to run the FAILURE_TESTS as it seems that we were not executing this test on a PR submission.

Boomatang reviewed (commented):
Most of this looks good but there is a large concern. 

The changes can leave the deployments scaled down if there is a failure in test which will leave the cluster in a bad state. This may have unexpected side affects on other tests. No matter how or when the test finishes the cluster should be returned to the same state as before the test started. 

This is only the code review and won't be running the code till the concern above is addressed.

MStokluska commented:
> Most of this looks good but there is a large concern.
> 
> The changes can leave the deployments scaled down if there is a failure in test which will leave the cluster in a bad state. This may have unexpected side affects on other tests. No matter how or when the test finishes the cluster should be returned to the same state as before the test started.
> 
> This is only the code review and won't be running the code till the concern above is addressed.

Thanks for review. I've addressed the comments and updated the PR, going to re-run rhoam-e2e and e2e to verify.

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test e2e

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test e2e

psturc commented:
/lgtm

matskiv commented:
/approve
